db/commitlog/commitlog.cc

```diff
@@ -2593,12 +2593,12 @@ db::commitlog::read_log_file(sstring filename, sstring pfx, commit_load_reader_f
             return eof || next == pos;
         }
         future<> skip(size_t bytes) {
-            pos += bytes;
-            if (pos > file_size) {
+            auto n = std::min(file_size - pos, bytes);
+            pos += n;
+            if (pos == file_size) {
                 eof = true;
-                pos = file_size;
             }
-            return fin.skip(bytes);
+            return fin.skip(n);
```

Comment:
Shouldn't we return an error in this case?
I'm concerned this could obscure corrupt entries.

Reply:
The only places skip is when we've already determined a chunk/entry is corrupted. So we are not missing anything, this is just guarding against whether the assumed size of the corrupt data is correct or not.

Reply:
Though techically we could throw here I guess, since there is no way we can continue anyway (no way to find any next chunk or whatnot). We can't really add to "corrupt size" though, since we don't know if we have truncation or borked data being read for skip dist.

Reply:
If a chunk is corrupt we cannot rely on its header and skip to the next chunk.
We might as well stop processing the commitlog segment at this point rather than trying to skip to the next chunk.

OTOH if we didn't detect that the chunk is corrupt before hand and its size is too large, then either the chunk is corrupt or the commitlog file got truncated.
Either way it should be considered an error IMO.
Silently "fixing" it with no trace seems wrong.